docs/changes/units/11786.feature.rst

```diff
@@ -0,0 +1,10 @@
+A new unit, ``redshift``, is defined. It is a dimensionless unit to
+distinguish redshift quantities from other non-redshift values. For compatibility
+with dimensionless quantities the equivalency ``dimensionless_redshift`` is added.
+This equivalency is enabled by default.
+
+Also added is the equivalency ``with_cosmological_redshift`` to enable
```

Comment:
Need to rewrite the docs a bit to reflect the change. Move to Cosmology folder, etc.